File: 125-valid-palindrome/solution.py
# Valid Palindrome
# https://leetcode.com/problems/valid-palindrome/

import re


class Solution:
    def isPalindrome(self, s: str) -> bool:
        cleaned_str = re.sub(r'[^a-zA-Z0-9]', '', s.lower())

        for i in range(len(cleaned_str) // 2):
            if cleaned_str[i] != cleaned_str[-1 - i]:
                return False

        return True

File: 125-valid-palindrome/solution.js
// Valid Palindrome
// https://leetcode.com/problems/valid-palindrome/

/**
 * @param {string} s
 * @return {boolean}
 */
var isPalindrome = function (s) {
  // const isAlphaNum = (c) => /[a-z0-9]/i.test(c);
  const str = s.toLowerCase().replaceAll(/[\W_]/g, "");

  for (let i = 0; i < Math.floor(str.length / 2); i++) {
    if (str[i] !== str[str.length - 1 - i]) return false;
  }

  return true;
};
